src/core/job.c

```diff
@@ -1332,6 +1333,56 @@ int job_deserialize(Job *j, FILE *f) {
         return 0;
 }
 
+static int job_unit_build_json(sd_json_variant **ret, const char *name, void *userdata) {
+        _cleanup_free_ char *p = NULL;
+        Job *j = ASSERT_PTR(userdata);
+
+        assert(ret);
+
+        p = unit_dbus_path(j->unit);
+        if (!p)
+                return -ENOMEM;
+
+        return sd_json_buildo(ret,
+                        SD_JSON_BUILD_PAIR_STRING("id", j->unit->id),
+                        SD_JSON_BUILD_PAIR_STRING("path", p));
```

Comment:
please, no. let's not expose the dbus path over varlink. that's a dbus thing, and really redundant. We should use unit names as identifiers, nothing else via varlink.